pytorch_lightning/callbacks/quantization.py

```diff
@@ -82,60 +82,65 @@ def _recursive_hasattr(obj: Any, attribs: str, state: bool = True) -> bool:
+    """
+    Quantization allows speeding up inference and decreasing memory requirements
+    by performing computations and storing tensors at lower bitwidths
+    (such as INT8 or FLOAT16) than floating point precision.
+    We use native PyTorch API so for more information
+    see `Quantization <https://pytorch.org/docs/stable/quantization.html#quantization-aware-training>`_.
+
+    .. warning:: ``QuantizationAwareTraining`` is in beta and subject to change.
 
-    def __init__(
-        self,
-        qconfig: Union[str, QConfig] = 'fbgemm',
-        observer_type: str = "average",
-        collect_quantization: Optional[Union[int, Callable]] = None,
-        modules_to_fuse: Optional[Sequence] = None,
-        input_compatible: bool = True,
-    ) -> None:
-        """
-        Quantization allows speeding up inference and decreasing memory requirements
-        by performing computations and storing tensors at lower bitwidths
-        (such as INT8 or FLOAT16) than floating point precision.
-        We use native PyTorch API so for more information
-        see `Quantization <https://pytorch.org/docs/stable/quantization.html#quantization-aware-training>`_.
 
-        .. warning:: ``QuantizationAwareTraining`` is in beta and subject to change.
+    Args:
 
+        qconfig: quantization configuration:
 
-        Args:
+            - 'fbgemm' for server inference.
+            - 'qnnpack' for mobile inference.
+            -  a custom `torch.quantization.QConfig
+                <https://pytorch.org/docs/stable/torch.quantization.html#torch.quantization.QConfig>`_.
 
-            qconfig: quantization configuration:
+        observer_type: allows switching between ``MovingAverageMinMaxObserver`` as "average" (default)
+            and ``HistogramObserver`` as "histogram" which is more computationally expensive.
 
-                - 'fbgemm' for server inference.
-                - 'qnnpack' for mobile inference.
-                -  a custom `torch.quantization.QConfig <https://pytorch.org/docs/stable/torch.quantization.html#torch.quantization.QConfig>`_.
+        collect_quantization: count or custom function to collect quantization statistics:
 
-            observer_type: allows switching between ``MovingAverageMinMaxObserver`` as "average" (default)
-                and ``HistogramObserver`` as "histogram" which is more computationally expensive.
+            - ``None`` (deafult). The quantization observer is called in each module forward
+                (useful for collecting extended statistic when useing image/data augmentation).
+            - ``int``. Use to set a fixed number of calls, starting from the beginning.
+            - ``Callable``. Custom function with single trainer argument.
+                See this example to trigger only the last epoch:
 
-            collect_quantization: count or custom function to collect quantization statistics:
+                .. code-block:: python
```

Comment:
I would avoid adding code blocks under individual arguments. If we do this then one needs to justify for each argument why there should or should not be a code sample. 
do you think we could move it to an example section below?

Reply:
good point, so move it to main or drop it?

Reply:
no preference :) what you prefer <3

Reply:
well this is quite specific, so thinking to keep it and rather add one more general one...